tests/platform_tests/test_intf_fec.py

```diff
@@ -179,3 +140,57 @@ def skip_ber_counters_test(intf_status: dict) -> bool:
             pytest.fail("Pre-FEC and Post-FEC BER are not valid floats for interface {}, \
                     fec_pre_ber: {} fec_post_ber: {}"
                         .format(intf_name, fec_pre_ber, fec_post_ber))
+
+
+def get_fec_histogram(duthost, intf_name):
+    """
+    @Summary: Fetch FEC histogram for a given interface.
+    """
+    try:
+        logging.info("Get output of 'show interfaces counters fec-histogram {}'".format(intf_name))
+        fec_hist = duthost.show_and_parse("show interfaces counters fec-histogram {}".format(intf_name))
+    except Exception as e:
+        logging.error("Failed to execute 'show interfaces counters fec-histogram {}': {}".format(intf_name, e))
+        pytest.fail("Command 'show interfaces counters fec-histogram {}' not found \
+                or failed: {}".format(intf_name, str(e)))
+        return None
+
+    if not fec_hist:
+        pytest.fail("No FEC histogram data found for interface {}".format(intf_name))
+
+    logging.info("FEC histogram for interface {}: {}".format(intf_name, fec_hist))
+    return fec_hist
+
+
+def validate_fec_histogram(duthost, intf_name):
+    """
+    @Summary: Validate FEC histogram critical bins for any errors. Fail the test if bin value > 0
+    """
+
+    fec_hist = get_fec_histogram(duthost, intf_name)
+    if not fec_hist:
+        pytest.fail("FEC histogram data not found for interface {}".format(intf_name))
+
+    critical_bins = range(7, 16)
+    for bin_index in critical_bins:
+        bin_value = int(fec_hist[bin_index].get('codewords', 0))
+        if bin_value > 0:
+            pytest.fail("FEC histogram bin {} has errors for interface {}: {}"
+                        .format(bin_index, intf_name, bin_value))
+
+
+def test_verify_fec_histogram(duthosts, enum_rand_one_per_hwsku_frontend_hostname,
+                              enum_frontend_asic_index, conn_graph_facts):
+    """
+    @Summary: Verify the FEC histogram is valid and check for errors
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+    if "broadcom" in duthost.facts.get('platform_asic'):
+        pytest.skip("Skipping this test on platforms with Broadcom ASICs")
+
+    # Get valid interfaces
+    valid_interfaces = get_valid_interfaces(duthost, SUPPORTED_SPEEDS)
+
+    for intf_name in valid_interfaces:
+        wait_until(30, 10, 0, validate_fec_histogram, duthost, intf_name)
```

Comment:
@vvolam don't understand the intention here especially given the `wait_until` definition

```
    @return: If the condition function returns True before timeout, return True. If the condition function raises an
        exception, log the error and keep waiting and polling.
```